homeassistant/components/soma/__init__.py

```diff
@@ -0,0 +1,116 @@
+"""Support for Soma Smartshades."""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant import config_entries
+from homeassistant.components.soma import config_flow
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import HomeAssistantType
+
+from homeassistant.const import (
+    CONF_HOST, CONF_PORT)
+
+from .const import DOMAIN, HOST, API
+
+DEVICES = 'devices'
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=10)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PORT): cv.string
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+SOMA_COMPONENTS = ['cover']
+
+
+async def async_setup(hass, config):
+    """Set up the Soma component."""
+    if DOMAIN not in config:
+        return True
+
+    hass.data[DOMAIN] = {}
```

Comment:
We can remove this.

Reply:
It seems I can safely remove everything in this function except for returning True. It seems the function itself has to be there but I don't need to do anything in it.

Reply:
We should keep the start of the config flow import step if there's a config section for this domain. If not, just return True.